src/private/cstr.rs

```diff
@@ -19,6 +19,16 @@ pub fn set_str(buf: &mut [u8], s: &str) -> Result<(), crate::sys::EspError> {
     Ok(())
 }
 
+pub fn set_cchar_slice(buf: &mut [c_char], s: &str) -> Result<(), EspError> {
```

Comment:
I don't think this is correct w.r.t. ASCIIZ strings. You are not procesing the `\0` byte correctly.

Also, rather than having yet another function that sets a `core::str` as an ASCIIZ C string, can you just use `set_str` from above? If necessary, please make it faillble rather than asserting. 

Also and if necessary, please make it working without requiring the `alloc` module.

Reply:
I don't think this is resolved. How is this whole function you introduced any different than `set_str`? Please fix `set_str` instead if it needs fixing (as in, make it not assert but return an error instead).

Reply:
Sorry, I did not realize that you are now calling `set_str`. Still, why is your function necessary? If it is just to re-cast `&mut [u8]` to `&mut [c_char]` can you please do it in the calling code instead?